src/common/values.ts

```diff
@@ -137,23 +138,27 @@ const writeValuesToFile = async (
 /**
  * Writes new values to the repo. Will keep the original values if `overwrite` is `false`.
  */
-export const writeValues = async (values: Record<string, any>, overwrite = false): Promise<void> => {
+export const writeValues = async (inValues: Record<string, any>, overwrite = false): Promise<void> => {
   const d = terminal('common:values:writeValues')
-  d.debug('Writing values: ', values)
-  hasSops = existsSync(`${env.ENV_DIR}/.sops.yaml`)
-  // We relocated teamConfig.teams to teamConfig, so one time we might have to fall back to our previous location
-  // TODO:[teamConfig] deprecate this fallback sometime in the future
-  const teams = Object.keys((values.teamConfig?.teams ?? values.teamConfig) || {}).filter((i) => i !== 'teams')
+  d.debug('Writing values: ', inValues)
+  hasSops = await pathExists(`${env.ENV_DIR}/.sops.yaml`)
+  const teamConfig = (await loadYaml(`${env.ENV_DIR}/env/teams.yaml`)) as Record<string, Record<string, any>>
```

Comment:
please add a comment in code that explains why teamConfig needs to be read from file while saving the inValues

Reply:
values might not be ready at that stage but the teamConfig is needed?

Reply:
should be self explanatory

Reply:
but it is not. I think adding comment like 
```
# values might not be ready at that stage but the teamConfig is needed
```

Would make the work